### DISCUSSION TITLE
學習新程式的步驟

### DISCUSSION THREAD

**Discussion by zhilin93517** [1101-GS4719-C/Python-Community-GS4719-C#2131] [General]
在學習一個全新python package的時候，你們喜歡從最基本的東西學完之後再往其他概念學，還是先大致的概念學過之後再繼續往更深的地方學?
我以前喜歡第一種，但發現越學越吃力，反到最後花了一堆時間卻沒有真的學到什麼

**Comment by Jing-Asteroid** [comment]
烙格沒事，其實大部分的套件都會有一些官網，如果真不行的話可以直接去官網那邊看相對應的代號或是程式是什麼意涵，只不過大部分都是英文的，所以要稍微自行翻譯一下。

**Comment by 112121py** [comment]
我傾向先學習大致的概念，但我不太確定這是不是你的意思（？，就可能先大概了解這個套件可以做什麼事，有哪些應用場景，做得出什麼樣的成果，然後再從簡單的需求學習到複雜的需求，相對能夠知道這個package 在做什麼，但其實我不太確定這是不是比較像第一種xd

**Comment by Jinyuu17** [comment]
從最基本的開始，逐步深入
優勢：這種方法適合於那些希望徹底理解一個 package 的人，並且能夠逐步掌握每一個功能的細節。先學習基本概念和基礎功能，然後隨著理解的加深，逐漸接觸進階功能和更複雜的應用。這種方式有助於鞏固知識並形成扎實的基礎，特別是當這個 package 涉及到許多基礎結構或概念時。

**Comment by Jinyuu17** [comment]
先學習大致概念，再逐步深入
優勢：這種方式適合那些希望快速掌握大致概念，並且馬上應用於特定任務的人。你可以先學習 package 的核心功能，快速上手使用，然後隨著需要再回過頭學習進階的部分。這種方式能夠快速讓你使用這個工具，並在實際使用中慢慢加深理解。

**Comment by Jinyuu17** [comment]
我覺得沒有標準答案，只有適不適合現在的你。有時候結合兩者的優勢可能是最好的學習方式：先抓住全局，再深入關鍵細節。你可以先了解 package 的整體結構，快速上手一些常用的功能，然後根據項目需求進一步深入學習。

**Comment by gigitilto** [comment]
我會先去找程式範例來學習，因為我覺得實作是可以比較快學會的方式，先知道它的作用以及用法，就可以比較快上手。在後面進一步的學習當中，我覺得可以在使用的時候遇到不知道該怎麼寫下去的時候再查詢相關用法，遇到問題後得到解決方法我印象會很深刻。

**Comment by yuan3x2** [comment]
我自己認為先了解一個工具的使用情境再理解運作原理，也就是利用Top-Down的學習方法，能夠幫助我快速掌握新知式的概念以及應用方法，或許原PO可以試試看這種學習方法

**Comment by lai1008** [comment]
我自己認為先了解一個工具的使用情境再理解運作原理。能夠幫助我快速掌握新知式的概念以及應用方法，或許可以試試看這樣了解新程式

**Comment by poyen0806** [comment]
我會先去知道該package的使用情境
等確定摸索完功能後
再去思考運作原理並看看能不能用組合運用在其他情境

這邊建議如果學得很吃力的話
可以去看別人實際寫的影片
了解該package各函式的使用情境
這樣會比較好瞭解

**Comment by ShuliANN** [comment]
我自己傾向先學習基礎知識再加上實作，不但可以‘在解決實際問題時更容易理解所涉及的原理，並減少依賴查閱資料的時間。更可以幫助你將抽象的知識具體化，當遇到問題時，會迫使你更深入思考學到的概念，從而加深對它們的理解。通過實作，你會學會如何靈活應用學到的知識來解決各種問題，這不僅增強你的學習效果，還能提高實際工作中的解決問題能力。
